app/models/api_key.rb

```diff
@@ -42,4 +59,8 @@ def other_enabled_scopes?
   def scope_presence
     errors.add :base, "Please enable at least one scope" unless enabled_scopes.any?
   end
+
+  def not_soft_deleted?
+    errors.add :base, "Key is invalid. Please delete and create a new one." if soft_deleted?
+  end
```

Comment:
Added a validation here but practically it should never be called. May be redundant?

Reply:
Some of the best validations are the ones you almost never need.